src/libraries/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableHashSet.cs

```diff
@@ -86,6 +86,19 @@ public static ImmutableHashSet<T> CreateRange<T>(IEqualityComparer<T>? equalityC
         /// <param name="items">The items to prepopulate.</param>
         /// <returns>The new immutable collection.</returns>
         public static ImmutableHashSet<T> Create<T>(params T[] items)
+        {
+            Requires.NotNull(items, nameof(items));
+
+            return Create((ReadOnlySpan<T>)items);
+        }
+
+        /// <summary>
+        /// Creates a new immutable collection prefilled with the specified items.
+        /// </summary>
+        /// <typeparam name="T">The type of items stored by the collection.</typeparam>
+        /// <param name="items">The items to prepopulate.</param>
+        /// <returns>The new immutable collection.</returns>
+        public static ImmutableHashSet<T> Create<T>(ReadOnlySpan<T> items)
         {
             return ImmutableHashSet<T>.Empty.Union(items);
```

Comment:
I know this predates your changes, but it strikes me as odd that we're defining a core factory method in terms of a "Union" operation. Not familiar with the Union implementation but it sounds like it might be introducing unnecessary checks on the LHS?

Reply:
Potentially.  Happy for us to address any overheads here separately.